certs/pki/vault.go

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+// Package pki wraps vault client
+package pki
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/vault/api"
```

Comment:
Does it make sense to completely remove the Vault API client? I had a pretty bad experience with using Vault API and the client they provide is quite useless. I see that you also have to do a lot of parsing manually. I'm asking this because Vault is a giant dependency, and the only reason to have it is this client, which is not particularly useful.